src/GraphQL/DataLoader/DataLoaderBase.cs

```diff
@@ -27,15 +27,15 @@ public async Task DispatchAsync(CancellationToken cancellationToken = default)
                 // If cancellation has been requested already,
                 // set the task to cancelled without calling FetchAsync()
                 tcs.TrySetCanceled();
-                return;
+                return Task.FromResult(0);
             }
 
             try
             {
                 var result = await FetchAsync(cancellationToken)
                     .ConfigureAwait(false);
 
-                tcs.SetResult(result);
+                return Task.Run(() => tcs.SetResult(result), cancellationToken);
```

Comment:
I know this is not optimal, so if we can find any non-dead-locking solution without `Task.Run`, I'm all for it!